src/accelerate/accelerator.py

```diff
@@ -2551,7 +2553,9 @@ def load_state(self, input_dir: str, **load_model_func_kwargs):
             map_location,
             **load_model_func_kwargs,
         )
-        custom_checkpoints = [f for f in os.listdir(input_dir) if "custom_checkpoint" in f]
+        custom_checkpoints = [
+            f for f in os.listdir(input_dir) if re.search(r"custom_checkpoint_[0-9]+\.pkl$", f) is not None
```

Comment:
```suggestion
            f for f in os.listdir(input_dir) if re.search(r"^custom_checkpoint_[0-9]+\.pkl$", f) is not None
```

Reply:
We want to include, not exclude as we want only files named with `custom_checkpoint_` here. Accelerate stores these during `save_state` using this convention: https://github.com/huggingface/accelerate/blob/main/src/accelerate/checkpointing.py#L205

Reply:
I think Sylvain's suggestion is perfect, no?

Reply:
Yes it is, that's from me misunderstanding carrot, thanks @stas00 :)

Reply:
ah, of course, `^` can be confusing as it's overloaded to mean NOT inside a pattern, as in `[^0-9]`

Reply:
Yes exactly 😅